RUFAS/biophysical/animal/reproduction/reproduction.py

```
@@ -519,6 +521,45 @@ def _set_cow_reproduction_program(
 
         return reproduction_data_stream
 
+    def _simulate_first_estrus(
+        self,
+        reproduction_data_stream: ReproductionDataStream,
+        start_day: int,
+        simulation_day: int,
+        estrus_note: str,
+        avg_estrus_cycle: float,
+        max_cycle_length: float = math.inf,
+    ) -> ReproductionDataStream:
+        """
+        Calculate and set first next estrus day for an heiferII.
+
+        Parameters
+        ----------
+        start_day : int
+            The day to begin the estrus cycle calculation.
+        simulation_day : int
+            The current simulation day.
+        estrus_note : str
+            Note explaining the reason for estrus simulation.
+        avg_estrus_cycle : float
+            Average length of the estrus cycle.
+        max_cycle_length : float, optional
+            Maximum allowable length for the estrus cycle, by default inf.
+
+        Returns
+        -------
+        ReproductionDataStream
+            Updated reproduction datastream after estrus simulation.
+        """
+        estrus_cycle = random.randint(1, floor(avg_estrus_cycle))
+        if abs(estrus_cycle) >= max_cycle_length:
```

I wouldn't expect the value of estrus_cycle to ever be negative in the update (before it was drawn from a normal distribution I think, so technically could have been?) -- so maybe don't need the "abs", though probably doesn't hurt anything

---

Then we should throw an error if it is ever negative.